### PR TITLE
serve on unix socket as default

### PR DESCRIPTION
**Pull request by ahalimx86**
- change IPU pluging to serve on unix socket as default to be compliant with DPU daemon
- fix golangci-lint timeout error

### CODE REVIEW

**Review by palade — COMMENTED** [review]
The implementation looks good to me, but one thought: would be possible to merge `servingAddr` and `servingProto` into one parameter instead of having two? I'm thinking that based on the format of the `servingAddr` should be enough to know which protocol is required

**Comment by ahalimx86** [comment]
> The implementation looks good to me, but one thought: would be possible to merge `servingAddr` and `servingProto` into one parameter instead of having two? I'm thinking that based on the format of the `servingAddr` should be enough to know which protocol is required

We could do that but that would require extra parsing and validation. With just `servingAddr` user would have to specify `unix://var/run/vendor-socket.sock` or `tcp://localhost:<port>` OR `tcp://192.168.1.1:<port>` something like that.
Right now these options are there ONLY for when user wants to use other address instead of the default unix socket.

File: ipu-plugin/ipuplugin/cmd/rootcmd.go [hunk 1]
```diff
@@ -32,9 +32,9 @@ import (
 const (
 	configFilePath      = "/etc/ipu/"
 	cliName             = "ipuplugin"
-	defaultServingHost  = "localhost"
+	defaultServingAddr  = "/var/run/vendor-plugin.sock"
```
Comment (Eoghan1232) [inline ipu-plugin/ipuplugin/cmd/rootcmd.go:35]: nit:
```suggestion
	defaultServingAddr  = "/var/run/ipuplugin.sock"
```
to be consistant - either we change to vendor-plugin or keep ipuplugin throughout, wdyt?

**Review by Eoghan1232 — APPROVED** [review]

File: ipu-plugin/ipuplugin/cmd/rootcmd.go (hunk 1 above)
Comment (ahalimx86) [reply ipu-plugin/ipuplugin/cmd/rootcmd.go:35]: I didn't pick the socket name. I used the same name as DPU daemon defined in [here](https://github.com/openshift/dpu-operator/blob/8dd172060f0b7d9cd4ef82b57f583d2dda56f54d/daemon/vendorplugin.go#L16)

**Pull request merged**